engines/bastion_katello/app/assets/javascripts/bastion_katello/content-views/details/views/content-view-versions.html

```diff
@@ -77,22 +77,40 @@
           </td>
           <td bst-table-cell class="preserve-newlines">{{ version.description }}</td>
           <td class="col-sm-2">
-            <button type="button" class="btn btn-default"
-              ng-click="$state.go('content-view.promotion', {contentViewId: contentView.id, versionId: version.id})"
-              ng-hide="denied('promote_or_remove_content_views', contentView)"
-              ng-disabled="taskInProgress(version) || taskFailed(version) || pendingVersionTask">
-              <span translate>
-                Promote
-              </span>
-            </button>
-            <button type="button" class="btn btn-default"
-              ng-click="$state.go('content-view.version-deletion.environments', {contentViewId: contentView.id, versionId: version.id})"
-              ng-hide="denied('promote_or_remove_content_views', contentView)"
-              ng-disabled="taskInProgress(version) || taskFailed(version) || pendingVersionTask">
-              <span translate>
-                Remove
-              </span>
-            </button>
+            <div class="btn-group">
+              <button type="button" class="btn btn-default"
+                ng-click="$state.go('content-view.promotion', {contentViewId: contentView.id, versionId: version.id})"
+                ng-hide="denied('promote_or_remove_content_views', contentView)"
+                ng-disabled="taskInProgress(version) || taskFailed(version) || pendingVersionTask">
+                <span translate>
+                  Promote
+                </span>
+              </button>              <button type="button" class="btn dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
+                <span class="caret"></span>
+                <span class="sr-only">Toggle Dropdown</span>
+              </button>
+              <ul class="dropdown-menu">
+                <li><a href="#">Promote</a></li>
+                <li>
+                  <a href="#"
+                     ng-click="regenerateRepositories(version)"
+                     ng-hide="denied('promote_or_remove_content_views', contentView)"
+                     ng-disabled="taskInProgress(version) || taskFailed(version) || pendingVersionTask"
+                     translate>
+                     Regenerate Repository metadata
```

Comment:
I don't think so, one does the action along side a promote/publish, this one does not require it (and performs it across all published 'copies' in different lifecycle environments).